domain/application/service/unit.go

```diff
@@ -226,6 +231,32 @@ func (s *Service) UpdateCAASUnit(ctx context.Context, unitName coreunit.Name, pa
 	return nil
 }
 
+// SetReportedUnitAgentVersion sets the reported agent version for the
+// supplied unit name. Reported agent version is the version that the agent
+// binary on this unit has reported it is running.
+//
+// The following errors are possible:
+// - [coreerrors.NotValid] - when the reportedVersion is not valid.
+// - [coreerrors.NotSupported] - when the architecture is not supported.
+// - [applicationerrors.UnitNotFound] - when the unit does not exist.
+// - [applicationerrors.UnitIsDead] - when the unit is dead.
+func (s *Service) SetReportedUnitAgentVersion(ctx context.Context, unitUUID coreunit.UUID, reportedVersion agentbinary.Version) error {
```

Comment:
This function needs to take a unit name instead of a unit uuid. The facade directly deals in names. We will then need to transfer this over to a uuid inside the service.

Reply:
See other funcs in this file that take a `coreunit.Name` parameter.